src/core/storage/storage.js

```diff
@@ -8,12 +11,59 @@
  */
 export default class GlobalStorage {
   constructor (stateUpdateListener, stateResetListener) {
+    /**
+     * The update listener for changes in state (persistent storage changes)
+     *
+     * @type {Function}
+     */
+    this.stateUpdateListener = stateUpdateListener || function () {};
```

Comment:
state is a bit of a loaded term, though it makes sense here, maybe something like `persistedStateUpdateListener`? my suggestion is a little wordy so open to ideas! maybe a `persistedStateListeners` object instead of 2 params?